runtime/util_core/j9shchelp.c

```diff
@@ -357,9 +387,16 @@ isCacheFileName(J9PortLibrary* portlib, const char* nameToTest, uintptr_t expect
 		}
 	}
 	nameToTestLen = strlen(nameToTest);
-	if ((nameToTest[nameToTestLen-3] != 'G') && (nameToTest[nameToTestLen-4] != '_')) {
+	/**
+	 * For old cache file with no layer number like C290M4F1A64P_CC1_G37, 'G' is at nameToTestLen-3.
+	 * For cache file with layer number like C290M4F1A64P_CC1_G37L01, 'G' is at nameToTestLen-6.
+	 */
+	if (((nameToTest[nameToTestLen-3] != 'G') && (nameToTest[nameToTestLen-4] != '_'))
+		&& ((nameToTest[nameToTestLen-6] != 'G') && (nameToTest[nameToTestLen-7] != '_'))
```

Comment:
For robustness, this should check for 'L' as well.

Reply:
I added a call of getLayerFromName() here and the check for "L" is done inside getLayerFromName(). 

I think the original check 
`(nameToTest[nameToTestLen-3] != 'G') && (nameToTest[nameToTestLen-4] != '_')`
is incorrect. I changed it to 
`(nameToTest[nameToTestLen-3] != 'G') || (nameToTest[nameToTestLen-4] != '_')`